### PR TITLE
Master -> Prod for 2703 only

### PR DESCRIPTION
**Pull request by briaguya**
Verifying new functionality by updating snippets:
Screen Shot 2019-09-11 at 3 12 16 AM

Verifying previous functionality by updating pages of every type:
Screen Shot 2019-09-11 at 3 17 21 AM

### CODE REVIEW

File: joplin/base/signals.py [hunk 1]
```diff
@@ -33,7 +33,7 @@ def generate_responsive_images(sender, **kwargs):
         image.get_rendition(f'width-{width}')
 
 
-def create_build_aws(instance, publish=False, request=None):
+def create_build_aws(content_type, instance, publish_action='edited', request=None):
```
Comment (Niicck) [inline joplin/base/signals.py]: Is `content_type` only used for logging in our slack messages?

File: joplin/base/signals.py [hunk 2]
```diff
@@ -187,15 +181,31 @@ def get_http_request():
         return None
 
 
+# By creating a signal reciever for each snippet model we have, we can avoid
+# needing to filter out large amounts of unwanted calls in our function logic
```
Comment (Niicck) [inline joplin/base/signals.py]: I don't understand that, but it sounds like a good thing. Do you mean a "large amounts of unwanted [create_build_aws] calls" or something else?

File: joplin/base/signals.py [hunk 3]
```diff
@@ -50,15 +50,9 @@ def create_build_aws(instance, publish=False, request=None):
     logger.debug("create_build_aws() Starting task")
 
     slack_message = ""
-    publish_action = ""
-
-    if (publish):
-        publish_action = "published"
-    else:
-        publish_action = "unpublished"
```
Comment (Niicck) [inline joplin/base/signals.py]: Was this unused code?

File: joplin/base/signals.py (hunk 1 above)
Comment (briaguya) [reply joplin/base/signals.py]: yes, this is only for slack messages, figured it'd be more clear this way considering we have "Recycle and Reuse drop off center" as a map, location, and contact

File: joplin/base/signals.py (hunk 3 above)
Comment (briaguya) [reply joplin/base/signals.py]: this used to be used for our slack message (either say we published or unpublished a page). I changed it so we just pass in a publish_action as a string for publish/unpublish, and have the default just be "edited"

File: joplin/base/signals.py (hunk 2 above)
Comment (briaguya) [reply joplin/base/signals.py]: when I had `@receiver(post_save)` in there without a specified sender I was getting all sorts of silly post_save events from django_session etc.

While I'd like to be able to make it more generic (for all snippets post save do x) having "if you're not all sorts of other django post_save stuff" logic in the fn seemed like a bad plan

**Pull request merged**